### PR TITLE
Astar.js Revamp

### PR DESCRIPTION
**Pull request by niklabh**

### CODE REVIEW

**Comment by hoonsubin** [comment]
Can you fix the unit test and update all the other PRs?

**Comment by niklabh** [comment]
> Can you fix the unit test and update all the other PRs?

Unit tests fixed

**Comment by niklabh** [comment]
@hoonsubin can you please check.

**Comment by niklabh** [comment]
@hoonsubin can you please review.

File: packages/api-derive/src/dappsStaking/index.ts [hunk 1]
```diff
@@ -8,6 +8,9 @@ import { ContractAddress } from '../types';
 
 export const getAddressEnum = (address: string) => ({ Evm: address });
 
+/**
+ * @deprecated stakers does not return results
```
Comment (hoonsubin) [inline packages/api-derive/src/dappsStaking/index.ts:12]: is there a replacement function for this? Or is this no longer needed completely?

Comment (niklabh) [reply packages/api-derive/src/dappsStaking/index.ts:12]: This doesn't work currently as well. Dapp staking pallet has stopped sending stakers.

File: packages/sdk-core/src/formatUnitAmount.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Remove the unnecessary decimals such as '.000' that comes from `<Balance>.toHuman()`
+ * @param amountWithUnit eg: '100.0000 SDN'
+ * @returns '100 SDN'
+ */
+export const formatUnitAmount = (amountWithUnit: string): string => {
```
Comment (hoonsubin) [inline packages/sdk-core/src/formatUnitAmount.ts:6]: I think this function is a bit strange. Why not convert `BN` value to string with proper formatting? Is the other way around ever used?

ex: 10000000000000000000 -> "10 SDN"

File: packages/sdk-core/src/reduceBalanceToDenom.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import BN from 'bn.js';
+
+export const ASTAR_SS58_FORMAT = 5;
+export const ASTAR_DECIMALS = 18;
+
+/**
+ * A helper function to convert the given node balance value into the given chain token decimal point as a string.
+ * EX: `femto` -> `PLM`
```
Comment (hoonsubin) [inline packages/sdk-core/src/reduceBalanceToDenom.ts:8]: Outdated ticker

**Review by hoonsubin — CHANGES_REQUESTED** [review]

File: packages/sdk-core/src/reduceBalanceToDenom.ts (hunk 3 above)
Comment (niklabh) [reply packages/sdk-core/src/reduceBalanceToDenom.ts:8]: what should be the latest

**Review by hoonsubin — APPROVED** [review]
@impelcrypto @bobo-k2 @sirius651 can you also review this too?

File: packages/sdk-core/src/formatUnitAmount.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  const words = amountWithUnit.split(' ');
+  const value = Number(words[0]);
```
Comment (sirius651) [inline packages/sdk-core/src/formatUnitAmount.ts:11]: what if `words` is blank array?

File: packages/sdk-core/src/units.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  arrUnitNames[defaultUnitIndex] = defaultName;
+};
+
+export const getUnitNames = () => {
+  return arrUnitNames;
+};
+
+export const getUnit = (unitType: string) => {
+  const index = arrUnitNames.findIndex((elem) => elem === unitType);
+  return arrUnitPrefixes[index];
```
Comment (bobo-k2) [inline packages/sdk-core/src/units.ts:30]: This will fail if unitType is not in array, index will be -1. You should limit unitType parameter values.

**Review by bobo-k2 — CHANGES_REQUESTED** [review]

File: packages/sdk-core/src/formatUnitAmount.ts (hunk 2 above)
Comment (impelcrypto) [reply packages/sdk-core/src/formatUnitAmount.ts:6]: @niklabh Thank you for working on this. 
You can remove this function from the library and I'll remove this function from our portal as well. I can fetch the amount and parse by `BN.toString()`

**Review by impelcrypto — APPROVED** [review]

File: packages/sdk-core/src/formatUnitAmount.ts (hunk 4 above)
Comment (niklabh) [reply packages/sdk-core/src/formatUnitAmount.ts:11]: handled

File: packages/sdk-core/src/units.ts (hunk 5 above)
Comment (niklabh) [reply packages/sdk-core/src/units.ts:30]: handled

**Review by bobo-k2 — APPROVED** [review]
LGTM

**Pull request merged**